### ISSUE TITLE
[feature request/1 min work] import Subset under namespace torch.utils.data

### ISSUE THREAD

**Issue by CDLuminate** [pytorch/pytorch#5922]
This is convenient when we need to manually split a validation set from a Dataset instance.

```patch
diff --git a/torch/utils/data/__init__.py b/torch/utils/data/__init__.py
index f3b008fb1..53da18468 100644
--- a/torch/utils/data/__init__.py
+++ b/torch/utils/data/__init__.py
@@ -1,3 +1,3 @@
 
-from .dataset import Dataset, TensorDataset, ConcatDataset
+from .dataset import Dataset, TensorDataset, ConcatDataset, Subset
 from .dataloader import DataLoader
```

**Comment by vadimkantorov** [comment]
also stumbled upon this

**Comment by zou3519** [comment]
Fixed by #7816
